Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -386,8 +387,14 @@ BoardInit (
     if (FeaturePcdGet (PcdSmbiosEnabled)) {
       InitializeSmbiosInfo ();
     }
-    if (FeatureCfgData != NULL && FeatureCfgData->Sio == 1){
-      SioInit();
+    if (FeatureCfgData != NULL){
+      if (FeatureCfgData->Sio == 1){
+        SioInit();
+      }
+      GetFipsModeData = NULL;
+      HeciGetFipsMode(GetFipsModeData);
+      DEBUG ((DEBUG_INFO, "HeciGetFipsMode = 0x%lx\n", *GetFipsModeData));
+      HeciSetFipsMode(FeatureCfgData->MeFipsMode);
     }
```

Comment:
Do HeciSetFipsMode if the state is different.

Reply:
Revised in latest patch